units/generate_tasks.go

```diff
@@ -156,21 +156,26 @@ func (j *generateTasksJob) generate(ctx context.Context, t *task.Task) error {
 }
 
 // handleError return mongo.ErrNoDocuments if another job has raced, the passed in error otherwise.
-func (j *generateTasksJob) handleError(v *model.Version, handledError error) error {
+func (j *generateTasksJob) handleError(pp *model.ParserProject, v *model.Version, handledError error) error {
 	if v == nil {
 		return handledError
 	}
 	versionFromDB, err := model.VersionFindOne(model.VersionById(v.Id).WithFields(model.VersionConfigNumberKey))
 	if err != nil {
 		return errors.Wrapf(err, "problem finding version %s", v.Id)
 	}
+	ppFromDB, err := model.ParserProjectFindOne(model.ParserProjectById(v.Id).WithFields(model.ParserProjectConfigNumberKey))
+	if err != nil {
+		return errors.Wrapf(err, "problem finding version %s", v.Id)
+	}
 	if versionFromDB == nil {
 		return errors.Errorf("could not find version %s", v.Id)
 	}
 	// If the config update number has been updated, then another task has raced with us.
 	// The error is therefore not an actual configuration problem but instead a symptom
 	// of the race.
-	if v.ConfigUpdateNumber != versionFromDB.ConfigUpdateNumber {
+	if v.ConfigUpdateNumber != versionFromDB.ConfigUpdateNumber ||
```

Comment:
Don't think this can ever happen, but does it make sense to proceed if our config # is newer than the config # in the db, meaning that we actually have the newer version?